src/rt/sections_elf_shared.d

```diff
@@ -522,8 +531,8 @@ version (Shared)
     void decThreadRef(DSO* pdso, bool decAdd)
     {
         auto tdata = findThreadDSO(pdso);
-        tdata !is null || assert(0);
-        !decAdd || tdata._addCnt > 0 || assert(0, "Mismatching rt_unloadLibrary call.");
```

Comment:
What do you think of using the style `condition || abort("message")`? `abort` already does the trick of auto-filling the file name and line number.

Reply:
I don't like it too much (and haven't seen it much in druntime/Phobos), that's why I decided to wrap it with an extra helper function.